pkg/nuctl/test/deploy_test.go

```diff
@@ -124,6 +127,61 @@ func (suite *DeployTestSuite) TestDeployWithMetadata() {
 	suite.Require().Contains(suite.outputBuffer.String(), "0099887766")
 }
 
+func (suite *DeployTestSuite) TestDeployDirectory() {
```

Comment:
Thinking about this - the test to this feature should reside alongside the tests that verify archive, path as url, etc: https://github.com/nuclio/nuclio/blob/master/pkg/processor/build/runtime/test/suite/suite.go

After the recent refactor the benefit will be that this test runs for all runtimes and this is more of a builder feature than nuctl. We should try to test features as close as possible to the affected module.

In terms of testing, i can't see how you can use an existing function and just add function.yaml because you will alter the behavior of previous tests (i recall speaking of this, but perhaps the refactor changed things). As such you'll need to write a single test that does something like:

```
func (suite *TestSuite) TestBuildDir() {
	deployOptions := suite.getDeployOptionsDir("json-parser-with-function-config")
	
	deployOptions.FunctionConfig.Spec.Runtime = ""

	suite.DeployFunctionAndRequest(deployOptions,
		&httpsuite.Request{
			RequestBody:          `{"a": 100, "return_this": "returned value"}`,
			ExpectedResponseBody: "returned value",
		})
}
```

The reason being:
1. `getDeployOptionsDir` populates runtime (as if it were provided by the user) - we don't want this, we want the runtime to be read from function.yaml
2. `json-parser-with-function-config` is a function with function-yaml, implemented for all runtimes